enhancements/ingress/admin-configured-ca-bundle.md

```diff
@@ -0,0 +1,242 @@
+
+## Summary
+
+This enhancement describes the proposal to allow an administrator of OpenShift
+to configure trusted Certificate Authorities(CA) to verify the service's 
+certificate by OpenShift Router for the re-encrypted traffic.
+
+## Motivation
+
+Openshift administrator when creating a route for a service with re-encrypt termination
+type requires to provide CA certificate when a service is configured to use certificate
+signed by a CA different from the OpenShift CA to verify the service's certificate.
+And when the same CA is used by the administrator for signing several services'
+certificates it is unnecessary to configure the same CA certificate content as destinationCA
+in every created route object. And is also tiresome task having to rotate the
+certificate when expired, compromised or for any other reason in each route object. Instead,
+if an administrator could define a ConfigMap with the list of CA certificates
+and OpenShift Ingress Controller used it along with OpenShift CA to create a common 
+CA bundle for verifying service's certificates it would make managing routes a bit simpler.
+
+### User Stories
+- As an OpenShift cluster administrator, I want to configure admin or the enterprise
+  CA certificate bundle which should be used for verifying service's certificate by 
+  the OpenShift router, instead of configuring the same as destination CA certificate 
+  in each of the created route object.
+- As an OpenShift engineer, I want to create a CA trust bundle for Ingress operator usage,
+  which includes the OpenShift service CA and the admin provided CA bundle as the default
+  CA for verifying the service's certificate when a destination CA certificate is not
+  configured for a route.
+- As an OpenShift engineer, I want to update the OpenShift router certificate cache whenever
+  the OpenShift service CA certificate or the admin configured CA certificate bundle is 
+  updated or deleted.
+
+### Goals
+
+- An OpenShift administrator can configure a CA certificate bundle and expect
+  that to be used for verifying the service's certificate by the OpenShift
+  router without defining the destination CA certificate in the route object
+  created for the service.
+- Openshift Router should be updated seamlessly when the OpenShift service
+  CA certificate or the administrator configured CA certificate bundles are modified.
+
+### Non-Goals
+
+- OpenShift administrator defined CA certificate bundle will be used only for the
+  traffic through OpenShift router and for the routes created with reencrypt
+  termination type, any other scenario is outside the scope of this proposal.
+
+## Proposal
+
+OpenShift Ingress Operator to create a CA trust bundle which will contain the
+OpenShift service CA certificate and the admin configured CA certificates
+bundle and makes it available to the OpenShift Router as the default destination
+CA bundle file to be used for verifying the service's certificate. And also
```

Comment:
How should the router associate the service to the cert, without necessarily using a field in the Route?  Not all re-encrypt routes will be using this new feature.  This is related to my comment about making sure this is optional, and defining what happens when there is both a service CA and a global CA.

Reply:
The user need not configure or provide a reference to the configmap in the Route object, the new CA bundle will be used by default by the router only when a destinationCA is not configured by the user.

Reply:
Please explain why it would not be sufficient for the admin to change the default CA.  They could do the same thing by just updating the default CA with their own with their own trusted CA bundles.

Reply:
Okay.  If you could please add a brief explanation that would be great.  Something like:  this design proposes concatenating the default CA and custom CA together for the purposes of a common CA for the route component only.  A question was raised as to why not just allow the user to concatenate them together into a default CA, and that was answered as: because this is for routes only and the default CA is used by more than just routes.

Reply:
comment incorporated.